addons/website/static/src/snippets/s_dynamic_snippet_carousel/dynamic_snippet_carousel.js

```diff
@@ -5,14 +5,167 @@ import { utils as uiUtils } from "@web/core/ui/ui_service";
 
 export class DynamicSnippetCarousel extends DynamicSnippet {
     static selector = ".s_dynamic_snippet_carousel";
+    dynamicContent = {
+        ...this.dynamicContent,
+        _root: {
+            ...this.dynamicContent._root,
+            "t-on-slide.bs.carousel": (ev) => this.onSlide(ev),
+            "t-on-slid.bs.carousel": () => this.togglePrevButton(),
+        },
+    };
 
     setup() {
         super.setup();
         this.templateKey = "website.s_dynamic_snippet.carousel";
+        this.offset = 0;
+        this.chunkSize = 16;
+        this.totalToFetch = 0;
+        this.hasMore = true;
+        this.fetchedData = [];
+        this.index = -1;
+        this.holder = this.el.querySelector(".s_dynamic_content_holder");
+        this.content = this.el.querySelector(".s_dynamic_snippet_content");
+    }
+
+    async willStart() {
+        this.totalToFetch = parseInt(this.el.dataset.numberOfRecords) || 0;
+        await super.willStart();
+    }
+
+    start() {
+        super.start();
+        this.carouselEl = this.el.querySelector(".carousel");
+        this.togglePrevButton();
+    }
+
+    /**
+     *
+     * @returns {number} The index of the slide containing the "active" class.
+     */
+    getActiveSlideIndex() {
+        const slides = this.carouselEl.querySelectorAll(".carousel-item");
+        return [...slides].findIndex((s) => s.classList.contains("active"));
+    }
+
+    /**
+     * @returns {"single" | "all"} Carousel scrolling mode.
+     */
+    getScrollMode() {
+        return this.el.classList.contains("o_carousel_multi_items") ? "single" : "all";
+    }
+
+    /**
+     * Show/hide the "Previous" arrow.
+     *
+     * Prevents users from navigating back from the first slide,
+     * since lazy loading only adds slides when moving forward.
+     */
+    togglePrevButton() {
+        const prev = this.el.querySelector(".carousel-control-prev");
```

Comment:
Why not simply keep d-none when it is rendered first time and then this method will be called each time on slide